--- a/src/gui_w32.c
+++ b/src/gui_w32.c
@@ -127,6 +127,9 @@ gui_mch_set_rendering_options(char_u *s)
 	    return FAIL;
     }
 
+    if (!gui.in_use)
+	return OK;
+
     /* Enable DirectX/DirectWrite */
     if (dx_enable)
     {
diff --git a/src/option.c b/src/option.c
--- a/src/option.c
+++ b/src/option.c
@@ -7410,7 +7410,7 @@ did_set_string_option(
 
 #if defined(FEAT_RENDER_OPTIONS)
     /* 'renderoptions' */
-    else if (varp == &p_rop && gui.in_use)
+    else if (varp == &p_rop)
     {
 	if (!gui_mch_set_rendering_options(p_rop))
 	    errmsg = e_invarg;
