diff --git a/package.json b/package.json
index 1f63046..295c9be 100644
--- a/package.json
+++ b/package.json
@@ -23,8 +23,11 @@
     "proxyquire": "^2.1.3",
     "source-map-support": "^0.5.19",
     "temp": "^0.9.1",
+    "ts-loader": "^8.0.2",
     "ts-node": "^8.10.2",
-    "typescript": "^3.9.7"
+    "typescript": "^3.9.7",
+    "webpack": "^4.44.1",
+    "webpack-cli": "^3.3.12"
   },
   "engines": {
     "node": ">=12.10.0"
@@ -45,7 +48,7 @@
     "url": "https://github.com/thinca/node-vimhelp.git"
   },
   "scripts": {
-    "build": "tsc",
+    "build": "webpack",
     "clean": "rm -fr coverage lib",
     "lint": "eslint src test",
     "test": "nyc --reporter=lcovonly --reporter=text mocha",
diff --git a/webpack.config.js b/webpack.config.js
new file mode 100644
index 0000000..8601d4b
--- /dev/null
+++ b/webpack.config.js
@@ -0,0 +1,41 @@
+const path = require("path")
+
+module.exports = {
+  target: "node",
+  mode: "none",
+  resolve: {
+    mainFields: ["module", "main"],
+    extensions: [".js", ".ts"],
+    alias: {
+      "@": path.resolve(__dirname, "src"),
+    },
+  },
+  entry: "./src/index.ts",
+  output: {
+    path: path.join(__dirname, "lib"),
+    filename: "index.js",
+    libraryTarget: "commonjs",
+  },
+  module: {
+    rules: [
+      {
+        test: /\.ts$/,
+        include: [path.resolve(__dirname, "src")],
+        use: [
+          {
+            loader: "ts-loader",
+            options: {
+              compilerOptions: {
+                sourceMap: true,
+              },
+            },
+          },
+        ],
+      },
+    ],
+  },
+  node: {
+    __dirname: false,
+    __filename: false,
+  },
+}
