diff --git a/autoload/taskwarrior/data.vim b/autoload/taskwarrior/data.vim
index 0a0b52f..b1f435a 100644
--- a/autoload/taskwarrior/data.vim
+++ b/autoload/taskwarrior/data.vim
@@ -44,6 +44,9 @@ function! taskwarrior#data#get_args(...)
 
     let prompt = key . ":"
     echon "\<CR>".prompt.expr
+    if !has("nvim")
+      redraw
+    endif
 
     " getting inputs
     while 1
@@ -78,6 +81,9 @@ function! taskwarrior#data#get_args(...)
       echon "\<cr>".prompt.expr
     endif
 
+    if !has("nvim")
+      redraw
+    endif
     endwhile
 
     if expr !~ '^[ \t]*$' || a:1 == 'modify'
