diff --git a/src/gui_w32.c b/src/gui_w32.c
index d7bbc0cfe..7b4ba2a97 100644
--- a/src/gui_w32.c
+++ b/src/gui_w32.c
@@ -6562,10 +6562,45 @@ gui_mch_draw_string(
  * Output routines.
  */
 
+static UINT_PTR	    flushTimerId = 0;
+static BOOL	    flushRequired = FALSE;
+static VOID CALLBACK _OnFlushTimer( HWND hwnd, UINT uMsg UNUSED, UINT_PTR idEvent, DWORD dwTime UNUSED);
+static void _flush(void);
+
 /* Flush any output to the screen */
     void
 gui_mch_flush(void)
 {
+    if (flushTimerId == 0)
+	_flush();
+    else
+    {
+	flushRequired = TRUE;
+	_RPT0(_CRT_WARN, "flush canceled\n");
+    }
+}
+
+    static VOID CALLBACK
+_OnFlushTimer(
+    HWND hwnd,
+    UINT uMsg UNUSED,
+    UINT_PTR idEvent,
+    DWORD dwTime UNUSED)
+{
+    KillTimer(s_textArea, idEvent);
+    flushTimerId = 0;
+    if (flushRequired)
+    {
+	flushRequired = FALSE;
+	_flush();
+    }
+}
+
+    static void
+_flush(void)
+{
+    _RPT0(_CRT_WARN, "flush done\n");
+
 #   if defined(__BORLANDC__)
     /*
      * The GdiFlush declaration (in Borland C 5.01 <wingdi.h>) is not a
@@ -6581,6 +6616,8 @@ gui_mch_flush(void)
 #endif
 
     GdiFlush();
+
+    flushTimerId = SetTimer(s_textArea, 15, (UINT)0, _OnFlushTimer);
 }
 
     static void
