diff --git a/internal/jsonrpc2/jsonrpc2.go b/internal/jsonrpc2/jsonrpc2.go
index e465d3ab..677935bc 100644
--- a/internal/jsonrpc2/jsonrpc2.go
+++ b/internal/jsonrpc2/jsonrpc2.go
@@ -312,7 +312,7 @@ func (c *Conn) run(ctx context.Context) error {
 			if request.IsNotify() {
 				c.log(Receive, request.ID, -1, request.Method, request.Params, nil)
 				// we have a Notify, forward to the handler in a go routine
-				c.handle(ctx, c, request)
+				go c.handle(ctx, c, request)
 			} else {
 				// we have a Call, forward to the handler in another go routine
 				reqCtx, cancelReq := context.WithCancel(ctx)
@@ -324,7 +324,7 @@ func (c *Conn) run(ctx context.Context) error {
 				}
 				c.handlingMu.Unlock()
 				c.log(Receive, request.ID, -1, request.Method, request.Params, nil)
-				c.handle(reqCtx, c, request)
+				go c.handle(reqCtx, c, request)
 			}
 		case msg.ID != nil:
 			// we have a response, get the pending entry from the map