diff --git a/internal/lsp/cmd/serve.go b/internal/lsp/cmd/serve.go
index 2bf79f79..60bd9dcf 100644
--- a/internal/lsp/cmd/serve.go
+++ b/internal/lsp/cmd/serve.go
@@ -12,6 +12,8 @@ import (
 	"io"
 	"log"
 	"net"
+	"net/http"
+	_ "net/http/pprof"
 	"os"
 	"path/filepath"
 	"strings"
@@ -30,6 +32,8 @@ type Serve struct {
 	Port    int    `flag:"port" help:"port on which to run gopls for debugging purposes"`
 	Address string `flag:"listen" help:"address on which to listen for remote connections"`
 
+	PprofServer string `flag:"pprof.server" help:"address to enable pprof server"`
+
 	app *Application
 }
 
@@ -68,6 +72,11 @@ func (s *Serve) Run(ctx context.Context, args ...string) error {
 		log.SetOutput(io.MultiWriter(os.Stderr, f))
 		out = f
 	}
+	if s.PprofServer != "" {
+		go func() {
+			log.Println(http.ListenAndServe(s.PprofServer, nil))
+		}()
+	}
 	if s.app.Remote != "" {
 		return s.forward()
 	}