diff --git a/src/completion/widget/fzf-or-normal-completion b/src/completion/widget/fzf-or-normal-completion
index bcd2d9f..d246d16 100644
--- a/src/completion/widget/fzf-or-normal-completion
+++ b/src/completion/widget/fzf-or-normal-completion
@@ -1,7 +1,7 @@
 #autoload

 if [[ $BUFFER[-1] == $FZF_COMPLETION_TRIGGER && $BUFFER[-2] == " " ]]; then
-  zle fzf-completion
+  zle expand-or-complete
   return
 fi