diff --git a/ldrize_next_for_google/ldrize_next_for_google.user.js b/ldrize_next_for_google/ldrize_next_for_google.user.js
index 9399748..fc0e216 100644
--- a/ldrize_next_for_google/ldrize_next_for_google.user.js
+++ b/ldrize_next_for_google/ldrize_next_for_google.user.js
@@ -7,7 +7,7 @@
 // @grant        GM_openInTab
 // ==/UserScript==
 
-const SELECTED_STYLE = "background: #EEEEEE;";
+const SELECTED_STYLE = "background: #EFF4F8; border: #C4E6F8 2px solid;";
 const PINNED_STYLE = "border-left: #E4645C 4px solid; padding-left: 4px;";
 
 const initialize = (cursor, target, searchResult, pinList) => {
@@ -22,7 +22,7 @@ const initialize = (cursor, target, searchResult, pinList) => {
         cursor += 1;
         target = searchResult[cursor];
         if (pinList.includes(target)) {
-          target.style = `${PINNED_STYLE} ${SELECTED_STYLE}`;
+          target.style = `${SELECTED_STYLE} ${PINNED_STYLE}`;
         } else {
           target.style = SELECTED_STYLE;
         }
@@ -41,7 +41,7 @@ const initialize = (cursor, target, searchResult, pinList) => {
         cursor -= 1;
         target = searchResult[cursor];
         if (pinList.includes(target)) {
-          target.style = `${PINNED_STYLE} ${SELECTED_STYLE}`;
+          target.style = `${SELECTED_STYLE} ${PINNED_STYLE}`;
         } else {
           target.style = SELECTED_STYLE;
         }
@@ -56,7 +56,7 @@ const initialize = (cursor, target, searchResult, pinList) => {
     }
     pinList.forEach((_target) => {
       if (pinList.includes(_target) && target === _target) {
-        _target.style = `${PINNED_STYLE} ${SELECTED_STYLE}`;
+        _target.style = `${SELECTED_STYLE} ${PINNED_STYLE}`;
       } else if (pinList.includes(_target)) {
         _target.style = PINNED_STYLE;
       } else if (target === _target) {
@@ -113,7 +113,7 @@ const initialize = (cursor, target, searchResult, pinList) => {
         cursor += 1;
         target = searchResult[cursor];
         if (pinList.includes(getLinkFromElement(target))) {
-          target.style = `${PINNED_STYLE} ${SELECTED_STYLE}`;
+          target.style = `${SELECTED_STYLE} ${PINNED_STYLE}`;
         } else {
           target.style = SELECTED_STYLE;
         }
@@ -122,7 +122,7 @@ const initialize = (cursor, target, searchResult, pinList) => {
       }
       pinList.forEach((_target) => {
         if (pinList.includes(_target) && target === _target) {
-          _target.style = `${PINNED_STYLE} ${SELECTED_STYLE}`;
+          _target.style = `${SELECTED_STYLE} ${PINNED_STYLE}`;
         } else if (pinList.includes(_target)) {
           _target.style = `${PINNED_STYLE}`;
         } else if (target === _target) {
