$ git --no-pager diff HEAD\^ -- webpack.config.js
diff --git a/webpack.config.js b/webpack.config.js
index 97ed3600..5d85a2d4 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -5,6 +5,7 @@ const ForkTsCheckerNotifierWebpackPlugin = require('fork-ts-checker-notifier-web
 const webpack = require('webpack');
 const SentryPlugin = require('@sentry/webpack-plugin');
 const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin');

 let gitCommitHash = 'local';
 if (process.env.CIRCLECI) {
@@ -40,11 +41,14 @@ const config = [
   `XXXXXXX_API_BASE_URL`,
 ].reduce((cfg, key) => ({ ...cfg, [key]: process.env[key] }), {});

+const babelAndTsLoader = [{ loader: 'babel-loader' }, { loader: 'ts-loader', options: { transpileOnly: true } }];
+const tsLoader = [{ loader: 'ts-loader', options: { transpileOnly: true } }];
+
 const webpackConfig = {
   entry: './src/browser.ts',
   output: {
     path: path.resolve(__dirname, './public'),
-    filename: '[name].js?[contenthash]',
+    filename: '[name].js?[hash]',
     chunkFilename: '[chunkhash].js?[contenthash]',
     publicPath: '/',
   },
@@ -52,11 +56,7 @@ const webpackConfig = {
     rules: [
       {
         test: /\.tsx?$/,
-        loader: 'ts-loader',
-        options: {
-          transpileOnly: true,
-          experimentalWatchApi: true,
-        },
+        use: process.env['USE_HMR'] === 'true' ? babelAndTsLoader : tsLoader,
       },
     ],
   },
@@ -83,6 +83,7 @@ const webpackConfig = {
       config,
       version: gitCommitHash,
     }),
+    ...(process.env['USE_HMR'] === 'true' ? [new ReactRefreshWebpackPlugin()] : []),
   ],
   devServer: {
     port: parseInt(process.env['PORT']) || 8000,
@@ -91,6 +92,7 @@ const webpackConfig = {
     https: process.env['HTTPS'] === 'false' ? false : true,
     historyApiFallback: true,
     disableHostCheck: true,
+    hot: process.env['USE_HMR'] === 'true',
   },
   watchOptions: {
     poll: parseInt(process.env['WEBPACK_WATCH_OPTION_POLL']) || 1000,
