diff --git a/src/quickfix.c b/src/quickfix.c
index 6157650..bbeb7fd 100644
--- a/src/quickfix.c
+++ b/src/quickfix.c
@@ -7623,7 +7623,7 @@ hgr_search_file(
                        FALSE,  // vis_col
                        NULL,   // search pattern
                        0,      // nr
-                       1,      // type
+                       'H',    // type
                        TRUE    // valid
                        ) == QF_FAIL)
            {