diff --git a/src/syntax.c b/src/syntax.c
index 8c23d20a3..37906ea6c 100644
--- a/src/syntax.c
+++ b/src/syntax.c
@@ -9447,7 +9447,7 @@ syn_list_header(
        if (got_int)
            return TRUE;
        msg_outtrans(HL_TABLE()[id - 1].sg_name);
-       endcol = 15;
+       endcol = 14;
     }
     else if (msg_col + outlen + 1 >= Columns)
     {
@@ -9471,6 +9471,7 @@ syn_list_header(
     /* Show "xxx" with the attributes. */
     if (!did_header)
     {
+       msg_putchar(' ');
        msg_puts_attr("xxx", syn_id2attr(id));
        msg_putchar(' ');
     }