--- pyport.h.orig	Thu Mar 14 22:21:08 2019
+++ pyport.h	Thu Mar 14 22:22:06 2019
@@ -3,7 +3,9 @@
 
 #include "pyconfig.h" /* include for defines */
 
-#include <inttypes.h>
+#if !defined(_MSC_VER) || _MSC_VER > 1600
+# include <inttypes.h>
+#endif
 
 /**************************************************************************
 Symbols and macros to supply platform-independent interfaces to basic