diff --git a/ldrize_next_for_google/ldrize_next_for_google.user.js b/ldrize_next_for_google/ldrize_next_for_google.user.js
index dd82918..5beba94 100644
--- a/ldrize_next_for_google/ldrize_next_for_google.user.js
+++ b/ldrize_next_for_google/ldrize_next_for_google.user.js
@@ -107,6 +107,7 @@ class Manager {
     } else {
       pinCards.forEach((item) => {
         window.open(item.link);
+        item.unpin();
       });
     }
   }
