diff --git a/src/completion/complete.ts b/src/completion/complete.ts
index 3fb4be7a..3d0f5f82 100644
--- a/src/completion/complete.ts
+++ b/src/completion/complete.ts
@@ -191,7 +191,7 @@ export default class Complete {
         if (asciiCharactersOnly && !/^[\x00-\x7F]*$/.test(word)) {
           continue
         }
-        if ((!item.dup || source == 'tabnine') && words.has(word)) continue
+        // if ((!item.dup || source == 'tabnine') && words.has(word)) continue
         if (removeDuplicateItems && !item.isSnippet && words.has(word) && item.line == undefined) continue
         let filterText = item.filterText || item.word
         item.filterText = filterText