diff --git a/main.go b/main.go
index 5e450c1..aa6c7a9 100644
--- a/main.go
+++ b/main.go
@@ -217,6 +217,7 @@ func checkSixel() bool {
 			break
 		}
 	}
+	fmt.Printf("b = %q\n", b)
 	if !supported {
 		return false
 	}
@@ -229,6 +230,10 @@ func checkSixel() bool {
 }
 
 func main() {
+	c := checkSixel()
+	fmt.Printf("checkSixel() = %v\n", c)
+	os.Exit(0)
+
 	var nlong int
 	var ncolumns int
 	var rinterval float64
