diff --git a/src/gui_w32.c b/src/gui_w32.c
index 119bcd579..3b1469fe8 100644
--- a/src/gui_w32.c
+++ b/src/gui_w32.c
@@ -37,6 +37,7 @@ static int s_directx_scrlines = 0;
 # define IS_ENABLE_DIRECTX() (s_directx_enabled && s_dwc != NULL)
 static int directx_enabled(void);
 static void directx_binddc(void);
+static proftime_T dx_tm;
 #endif
 
 #ifdef FEAT_MENU
@@ -1681,7 +1682,12 @@ gui_mch_invert_rectangle(
 
 #if defined(FEAT_DIRECTX)
     if (IS_ENABLE_DIRECTX())
+    {
+	profile_start(&dx_tm);
 	DWriteContext_Flush(s_dwc);
+	profile_end(&dx_tm);
+	Trace("%s (%s)\n", profile_msg(&dx_tm), "gui_mch_invert_rectangle");
+    }
 #endif
 
     /*
@@ -1714,7 +1720,12 @@ gui_mch_draw_hollow_cursor(guicolor_T color)
 
 #if defined(FEAT_DIRECTX)
     if (IS_ENABLE_DIRECTX())
+    {
+	profile_start(&dx_tm);
 	DWriteContext_Flush(s_dwc);
+	profile_end(&dx_tm);
+	Trace("%s (%s)\n", profile_msg(&dx_tm), "gui_mch_draw_hollow_cursor");
+    }
 #endif
 
     /*
@@ -3054,7 +3065,12 @@ gui_mch_flash(int msec)
 
 #if defined(FEAT_DIRECTX)
     if (IS_ENABLE_DIRECTX())
+    {
+	profile_start(&dx_tm);
 	DWriteContext_Flush(s_dwc);
+	profile_end(&dx_tm);
+	Trace("%s (%s)\n", profile_msg(&dx_tm), "gui_mch_flash");
+    }
 #endif
 
     /*
@@ -3144,7 +3160,12 @@ gui_mch_delete_lines(
 	    use_redraw = 1;
 	}
 	else
+	{
+	    profile_start(&dx_tm);
 	    DWriteContext_Flush(s_dwc);
+	    profile_end(&dx_tm);
+	    Trace("%s (%s)\n", profile_msg(&dx_tm), "gui_mch_delete_lines");
+	}
     }
     if (!use_redraw)
 #endif
@@ -3194,7 +3215,12 @@ gui_mch_insert_lines(
 	    use_redraw = 1;
 	}
 	else
+	{
+	    profile_start(&dx_tm);
 	    DWriteContext_Flush(s_dwc);
+	    profile_end(&dx_tm);
+	    Trace("%s (%s)\n", profile_msg(&dx_tm), "gui_mch_insert_lines");
+	}
     }
     if (!use_redraw)
 #endif
@@ -6600,7 +6626,12 @@ gui_mch_flush(void)
 
 #if defined(FEAT_DIRECTX)
     if (IS_ENABLE_DIRECTX())
+    {
+	profile_start(&dx_tm);
 	DWriteContext_Flush(s_dwc);
+	profile_end(&dx_tm);
+	Trace("%s (%s)\n", profile_msg(&dx_tm), "gui_mch_flush");
+    }
 #endif
 
     GdiFlush();
@@ -8514,7 +8545,12 @@ gui_mch_drawsign(int row, int col, int typenr)
 
 #if defined(FEAT_DIRECTX)
     if (IS_ENABLE_DIRECTX())
+    {
+	profile_start(&dx_tm);
 	DWriteContext_Flush(s_dwc);
+	profile_end(&dx_tm);
+	Trace("%s (%s)\n", profile_msg(&dx_tm), "gui_mch_drawsign");
+    }
 #endif
 
     x = TEXT_X(col);
@@ -8998,7 +9034,12 @@ netbeans_draw_multisign_indicator(int row)
 
 #if defined(FEAT_DIRECTX)
     if (IS_ENABLE_DIRECTX())
+    {
+	profile_start(&dx_tm);
 	DWriteContext_Flush(s_dwc);
+	profile_end(&dx_tm);
+	Trace("%s (%s)\n", profile_msg(&dx_tm), "netbeans_draw_multisign_indicator");
+    }
 #endif
 
     for (i = 0; i < gui.char_height - 3; i++)
